src/sql/mir.rs

```diff
@@ -761,6 +761,28 @@ impl SqlToMirConverter {
                 }
             }
 
+
+            // 3. Get columns used by each filter node.
+            let mut filter_columns = HashMap::new();
+            let mut filter_nodes = Vec::new();
+            let mut moved_filters = Vec::new();
+            let mut sorted_rels: Vec<&String> = qg.relations.keys().collect();
+            sorted_rels.sort();
+            for rel in &sorted_rels {
+                let qgn = &qg.relations[*rel];
+                if *rel != "computed_columns" {
+                    if !qgn.predicates.is_empty() {
```

Comment:
That code looks different from your though. It doesn't just iterate over the predicates.